gnomad_hail/utils/variant_qc.py

```diff
@@ -262,80 +275,481 @@ def quantiles_to_bin_boundaries(quantiles: List[int]) -> Dict:
+    - singletons
+    - biallelics
+    - biallelic singletons
+    - adj
+    - adj biallelics
+    - adj singletons
+    - adj biallelic singletons
+
+    :param Table ht: Input table
+    :param int n_bins: Number of bins to bin into
+    :param bool singleton: Should bins be stratified by singletons
+    :param bool biallelic: Should bins be stratified by bi-alleleic variants
+    :param bool adj: Should bins be stratified by adj filtering
+    :param dictionary of boolean expr keyed by str add_substrat: Any additional stratifications for adding bins
+    :return table with bin number for each variant
+    :rtype: Table
+    """
+    def update_bin_expr(
+            bin_expr: Dict[str, hl.expr.BooleanExpression],
+            new_expr: hl.expr.BooleanExpression,
+            new_id: str
+    ) -> Dict[str, hl.expr.BooleanExpression]:
+        """
+        Updates a dictionary of expressions to add another stratification
+
+        :param Dict of BooleanExpressions keyed by strings bin_expr: Dictionary of expressions to add another
+        stratification to
+        :param BooleanExpression new_expr: New Boolean expression to add to `bin_expr`
+        :param str new_id: Name to add to each current key in `bin_expr` to indicate the new stratification
+        :return: Dictionary of `bin_expr` updated with `new_expr` added as an additional stratification to all
+        expressions already in `bin_expr`
+        :rtype: Dict of BooleanExpressions keyed by strings
+        """
+        bin_expr.update({
+            f'{new_id}_{bin_id}': bin_expr & new_expr
+            for bin_id, bin_expr in bin_expr.items()
+        })
+        return bin_expr
+
+    ht = ht.annotate(
+        singleton=ht.ac_raw == 1,
+        snv=hl.is_snp(ht.alleles[0], ht.alleles[1]),
+
+    )
+
+    ht = ht.filter(
+        ht.ac_raw > 0
+    ).persist()
+
+    # Desired bins and sub-bins
+    bin_expr = {
+        'bin': True
+    }
+
+    if singleton:
+        bin_expr = update_bin_expr(bin_expr, ht.singleton, "singleton")
+
+    if biallelic:
+        bin_expr = update_bin_expr(bin_expr, ~ht.was_split, "biallelic")
+
+    if adj:
+        bin_expr = update_bin_expr(bin_expr, (ht.ac > 0), "adj")
+
+    if add_substrat:
+        for add_id, add_expr in add_substrat.items():
+            bin_expr = update_bin_expr(bin_expr, add_expr, add_id)
+
+    bin_ht = compute_quantile_bin(
+        ht,
+        score_expr=ht.score,
+        bin_expr=bin_expr,
+        n_bins=n_bins
+    )
+    ht = ht.join(bin_ht, how='left')
+
+    return ht
+
+
+def default_score_bin_agg(
+        ht: hl.Table,
+        truth_ht: hl.Table,
+        fam_stats_ht: hl.Table
+) -> Dict[str, hl.expr.Aggregation]:
+    """
+    Default aggregation function to pass to `compute_aggregate_binned_data` to add aggregations for number of ClinVar
+    variants, number of truth variants (omni, mills, hapmap, and kgp_phase1), and family statistics.
+
+    Note that the following fields should be present:
```

Comment:
I don't really understand why, but Sphinx insists on a blank line between this line and "In ht:".